src/zh/cartoonmad/src/eu/kanade/tachiyomi/extension/zh/cartoonmad/CartoonMad.kt

```diff
@@ -0,0 +1,151 @@
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.asResponseBody
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+
+class CartoonMad : ParsedHttpSource() {
+    override val baseUrl = "https://www.cartoonmad.com"
+    override val lang: String
+        get() = "zh"
+    override val name: String
+        get() = "動漫狂"
+    override val supportsLatest: Boolean
+        get() = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .addInterceptor(::handleCharsetInterceptor)
+        .build()
+
+    fun handleCharsetInterceptor(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val response = chain.proceed(request)
+        if (request.url.encodedPath.contains("/comic/")) {
+            // Need an explicit definition of the charset format for the response
+            return response.newBuilder()
+                .body(
+                    response.body.source()
+                        .asResponseBody("text/html; charset=big5hkscs".toMediaType()),
+                )
+                .build()
+        }
+        return response
+    }
+
+    override fun chapterFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun chapterListSelector() = throw UnsupportedOperationException()
+
+    override fun imageUrlParse(document: Document) = throw UnsupportedOperationException()
+
+    override fun latestUpdatesFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/m/?act=1")
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        val content = document.select(content_selector)
+        return SManga.create().apply {
+            content.select("> table:nth-child(1) > tbody").let {
+                author =
+                    it.select("> tr > td:contains(作者：)").text().substringAfter("作者：").trim()
+                genre =
+                    it.select("> tr > td:contains(分類：) td:has(img[src=/image/start.gif])").text()
+                        .substringAfter("分類：").trim()
+                thumbnail_url =
+                    it.select("span.cover + img, span.covers + img").attr("abs:src")
+                        .ifEmpty { thumbnail_url }
+            }
+            description = content.select("> table:nth-child(2) legend + table").text()
+        }
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val doc = response.asJsoup()
+        val content = doc.select(content_selector)
+        val chapters = content.select("> table:nth-child(3) legend + table > tbody > tr > td > a")
+        return chapters.map {
+            SChapter.create().apply {
+                name = it.text()
+                url = it.attr("href")
```

Comment:
Use `setUrlWithoutDomain()` and `it.attr("abs:href")` to leave the responsibility of parsing relative paths to Jsoup.